docs/technical_tutorial/apollo_vehicle_adaption_tutorial_cn.md

```diff
@@ -0,0 +1,935 @@
+
+加速踏板位置：需准确反馈当前车辆实际踏板位置。
+
+纵向加速度：需准确反馈当前车辆的纵向加速度。
+
+车速：需准确反馈当前车辆车速，Apollo程序内需求的车速是m/s，**一般车辆反馈的车速单位是km/h，此时需要在DBC转换为适配代码后单独进行单位转化**。
+
+轮速：对于乘用车，需提供相关轮速信号，轮速大小，单位km/h。如果有条件，可提供车轮运动方向反馈，Value Description按照以下顺序0x00，FORWARD；0x01，BACKWARD；0x02，STANDSTILL；0x03，INVALID。
+
+发动机/电机转速：需准确反馈当前发动机或电机转速，单位r/min。
+
+故障信息：反馈驱动系统是否存在故障。
+
+4. 人工接管：
+
+（1）当车辆底盘接收到人工干预的加速踏板指令（或对于底盘小车，接收到遥控器下发的加速信号）时，车辆各个模块均退出自动驾驶。转向接管需要独立，此时驱动驾驶模式反馈被人工接管模式，转向驾驶模式、制动驾驶模式此时应反馈手动驾驶模式；
+
+（2）驱动自动驾驶被人工接管时，所有模块都退出自动驾驶模式。
+
+5. 越界处理
+
+发生越界时，转向需要退出自动驾驶模式。如定义目标加速踏板位置范围为0~100%，若Apollo下发的控制踏板指令不在此范围时，底盘需要对此越界拒绝执行，并退出自动驾驶模式。
+
+#### 线控制动
+
+1. 制动使能：
+
+（1）制动使能信号是控制车辆进入或退出驱动自动驾驶模式，该使能信号必须是通过上升沿触发方式的信号，使能信号原则上是1个信号；有些开发者若有不同信号同时控制，才能使能转向进入自动驾驶状态，需要开发者在底层自己集成相关使能信号的逻辑控制，给Apollo使能信号原则上必须只有1个。
+
+（2）制动使能信号必须独立，不能与转向模块使能信号相关联，需要满足：单独使能进入纵向自动驾驶模式，转向所处驾驶模式不发生变化，下发制动控制信号，车辆能够响应相应的控制命令；单独使能退出驱动自动驾驶模式，转向所处驾驶模式不发生变化，下发制动控制信号，车辆不响应；
+
+（3）原则上驱动使能信号与驱动、档位、驻车使能相互独立。根据以往的实际经验，有些车辆在纵向驱动和制动使能信号归为一个信号，这样的特殊情况需要开发者额外注意，确保纵向的使能一定要与横向使能分开并独立，纵向各模块如果相互关联，则使能需要驱动、制动、档位、驻车一同进去自动驾驶模式，不使能一同退出自动驾驶模式。
+
+（4）如果横向被接管，再次进入自动驾驶模式时，需要通过上升沿触发信号进入自动驾驶。
+
+2. 制动控制信号
+
+制动踏板目标位置：控制量是加速踏板的位置百分比，范围是0~100%，100%代表制动踏板踩到最大位置。
+
+目标减速度：可选项，控制量是车辆的目标减速度大小。
+
+制动灯控制：可选项，控制制动灯是否点亮，一般车辆的底盘会在制动时默认点亮制动灯。
+
+3. 制动反馈信号
+
+驾驶模式：反馈车辆当前所处的制动驾驶模式，至少需要3种模式：自动驾驶模式，手动驾驶模式，被人工接管模式。 
+
+制动踏板位置：制动踏板位置，百分比。
+
+刹车灯状态：反馈刹车灯是否点亮。
+
+故障信息：反馈制动系统是否有故障。
+
+4. 人工接管
+
+（1）当车辆底盘接收到人工干预的制动踏板指令（或对于底盘小车，接收到遥控器下发的减速信号）时，车辆各个模块均退出自动驾驶。转向接管需要独立，此时驱动驾驶模式反馈被人工接管模式，转向驾驶模式、制动驾驶模式此时应反馈手动驾驶模式；
+
+（2）驱动自动驾驶被人工接管时，所有模块都退出自动驾驶模式。
+
+5. 越界处理
+
+发生越界时，转向需要退出自动驾驶模式。如定义目标制动踏板位置范围为0~100%，若Apollo下发的控制踏板指令不在此范围时，底盘需要对此越界拒绝执行，并使所有模式退出自动驾驶模式。
+
+#### 线控档位
+
+1. 档位使能
+
+使能独立，不与横向控制相关联。可以与纵向使能信号关联，也可独立使能。档位使能信号必须是通过上升沿触发方式的信号，使能信号原则上是1个信号。
+
+2. 档位控制
+
+控制档位信号，档位信号必须按照以下顺序：0x00：N档；0x01：D；0x02：R；0x03：P；0x04：NONE。多余的不必再增加。
+
+3. 档位反馈
+
+档位信息：反馈当前档位信号，反馈顺序与控制档位信号顺序一致：0x00：N档；0x01：D；0x02：R；0x03：P；0x04：NONE。多余的不必再增加。
+
+故障信息：反馈换挡系统是否存在故障。
+
+#### 线控驻车
+
+1. 驻车使能
+
+使能独立，不与横向控制相关联。可以与纵向使能信号关联，也可独立使能。驻车使能信号必须是通过上升沿触发方式的信号，使能信号原则上是1个信号。
+
+2. 驻车控制
+
+驻车信号有效，电子手刹抱死，车辆驻车；驻车信号无效，电子手刹松开，车辆不驻车。
+
+3. 驻车反馈
+
+EPB开关状态：反馈当前电子手刹是否抱死或松开。
+
+驻车状态反馈：反馈当前驻车是自动驾驶模式还是手动控制模式。至少需要2种模式：自动驾驶模式，手动驾驶模式，被人工接管模式。
+
+驻车系统故障反馈：反馈驻车系统故障。
+
+#### 其他车身功能
+
+包括线控灯光，主要控制远、近光灯、转向灯、危险警报灯的打开和关闭，已经相应的状态反馈信号。
+
+还有雨刮和喇叭控制，作为可选项。
+
+#### 车辆VIN码
+
+VIN码一般17位，按照ASCII码格式，每一个ASCII占1字节，需要3帧报文连续发出，但是VIN码不需要实时更新，所以在系统请求进入自动驾驶时，VIN码通过CAN总线发出，并一直保持该值不再更新，也可减少总线的负载。
+
+### DBC文件要求
+
+熟悉了上述Apollo对车辆底盘信号的要求，第二步是进行车辆底盘信号database（DBC）文件进行编辑，设置通信的网络结构，每个信号的初值、符号类型，精度，大小范围，取值等，进而组合成相应的CAN通信报文（message）与Apollo进行通信。下面使用CANdb++软件对DBC文件进行编辑，有较好的可视化界面，该软件目前只适用于Windows系统。
+
+因为DBC文件后面会根据Apollo的转译脚本工具，将底盘定义的报文（message）、信号（signal）转化为C++程序代码，因此在编辑DBC时，对信号的名称定义、注释、赋值等就要符合C++语言定义规范，这样以确保在后期调试时不会因为DBC文件的问题无法调通CANBUS通信。根据Apollo代码要求，我们总结了以下注意事项：
+#### 1. 控制信号名称建议为 ACU
+在定义网络上ECU名称时，建议定义Apollo端的控制器名称为ACU（Apollo Control Unit）。
+
+![name](images/vehicle_adaption_tutorial/name.jpg)
+#### 2. CAN信号ID不大于2048
+目前乘用车CAN通信ID一般采用标准帧格式（CAN Standard），信号格式满足ISO11898的要求。
```

Comment:
@macDure Is it still valid? We recently added support for exteneded can frame. --- Only on ESD can if  through.